internal/output/html.go

```diff
@@ -0,0 +1,486 @@
+	FixedVersion     string
+	SeverityRating   string
+	SeverityScore    string
+}
+
+type HTMLVulnCount struct {
+	Critical int
+	High     int
+	Medium   int
+	Low      int
+	Unknown  int
+	Called   int
+	Uncalled int
+	Fixed    int
+	UnFixed  int
+}
+
+const UNFIXED = "No fix available"
+
+// HTML templates directory
+const TEMPLATEDIR = "html/*"
+
+// supportedBaseImages lists the supported OS base images for container scanning.
+var baseImages = []string{"Debian", "Alpine", "Ubuntu"}
+
+//go:embed html/*
+var templates embed.FS
+
+var vulnIndex = 0
+
+// BuildHTMLResults builds HTML results from vulnerability results.
+func BuildHTMLResults(vulnResult *models.VulnerabilityResults) HTMLResult {
+	var ecosystemMap = make(map[string][]SourceResult)
+	var resultCount HTMLVulnCount
+
+	for _, packageSource := range vulnResult.Results {
+		sourceName := packageSource.Source.String()
+		if strings.Contains(sourceName, "/usr/lib/") {
+			continue
+		}
+
+		// Process vulnerabilities for each source
+		sourceResult := processSource(packageSource)
+		if sourceResult == nil {
+			continue
+		}
+
+		sourceList, ok := ecosystemMap[sourceResult.Ecosystem]
+		if ok {
+			ecosystemMap[sourceResult.Ecosystem] = append(sourceList, *sourceResult)
+		} else {
+			ecosystemMap[sourceResult.Ecosystem] = []SourceResult{*sourceResult}
+		}
+
+		updateCount(&resultCount, &sourceResult.HTMLVulnCount)
+	}
+
+	// Build the final result
+	return buildHTMLResult(ecosystemMap, resultCount)
+}
+
+// processSource processes a single source (lockfile or artifact) and returns an SourceResult.
+func processSource(packageSource models.PackageSource) *SourceResult {
+	var allVulns []HTMLVulnResult
+	var calledPackages = make(map[string]bool)
+	var uncalledPackages = make(map[string]bool)
+	var uncalledVulnIDs = make(map[string]bool)
+	var groupIDs = make(map[string]models.GroupInfo)
+	ecosystemName := ""
+
+	for _, vulnPkg := range packageSource.Packages {
+		if ecosystemName == "" {
+			ecosystemName = vulnPkg.Package.Ecosystem
+		}
+
+		// Process vulnerability groups and IDs to get called/uncalled information
+		processVulnerabilityGroups(groupIDs, vulnPkg, uncalledVulnIDs, calledPackages, uncalledPackages)
+		// Process vulnerabilities from one source package
+		allVulns = append(allVulns, processVulnerabilities(vulnPkg)...)
+	}
+
+	// Split vulnerabilities into called and uncalled.
+	// Only add one vulnerability per group
+	packageResults := processPackageResults(allVulns, groupIDs, uncalledVulnIDs, ecosystemName)
+	var count HTMLVulnCount
+	for index, packageResult := range packageResults {
+		packageResults[index].Ecosystem = ecosystemName
+		packageResults[index].Source = packageSource.Source.Path
+		updateCount(&count, &packageResult.HTMLVulnCount)
+	}
+
+	return &SourceResult{
+		Source:         packageSource.Source.String(),
+		Ecosystem:      ecosystemName,
+		PackageResults: packageResults,
+		PackageCount:   [2]int{len(calledPackages), len(uncalledPackages)},
```

Comment:
Split this out into two variables, as it is a lot easier to keep track of which one is called and which is uncalled by explicitly naming them.